File: app/src/test/java/com/mapmate/domain/model/AppSettingsTest.java
package com.mapmate.domain.model;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;

import org.junit.Test;

public class AppSettingsTest {
    @Test
    public void defaults_matchRoutineRegistrationDefaults() {
        AppSettings settings = new AppSettings(
                AppSettings.DEFAULT_PERSONAL_BUFFER_MINUTES,
                AppSettings.DEFAULT_SAFETY_MARGIN_MINUTES,
                AppSettings.DEFAULT_NOTIFICATIONS_ENABLED,
                TransportMode.TRANSIT
        );

        assertEquals(6, settings.getPersonalBufferMinutes());
        assertEquals(5, settings.getSafetyMarginMinutes());
        assertFalse(settings.getNotificationsEnabled());
        assertEquals(TransportMode.TRANSIT, settings.getDefaultTransportMode());
    }

    @Test
    public void isValidBufferMinutes_acceptsOnlyZeroToSixty() {
        assertTrue(AppSettings.Companion.isValidBufferMinutes(0));
        assertTrue(AppSettings.Companion.isValidBufferMinutes(60));

        assertFalse(AppSettings.Companion.isValidBufferMinutes(-1));
        assertFalse(AppSettings.Companion.isValidBufferMinutes(61));
    }

    @Test
    public void bufferMinutesOrDefault_fallsBackForMissingOrInvalidValues() {
        assertEquals(12, AppSettings.Companion.bufferMinutesOrDefault(12, 6));
        assertEquals(6, AppSettings.Companion.bufferMinutesOrDefault(null, 6));
        assertEquals(6, AppSettings.Companion.bufferMinutesOrDefault(90, 6));
    }

    @Test
    public void transportModeOrDefault_fallsBackForUnknownNames() {
        assertEquals(TransportMode.WALK, AppSettings.Companion.transportModeOrDefault("WALK"));
        assertEquals(TransportMode.TRANSIT, AppSettings.Companion.transportModeOrDefault("BUS"));
        assertEquals(TransportMode.TRANSIT, AppSettings.Companion.transportModeOrDefault(null));
    }
}

File: app/src/test/kotlin/com/mapmate/domain/model/AppSettingsTest.kt
package com.mapmate.domain.model

import org.junit.Assert.assertEquals
import org.junit.Assert.assertFalse
import org.junit.Assert.assertTrue
import org.junit.Test

class AppSettingsTest {
    @Test
    fun defaults_matchRoutineRegistrationDefaults() {
        val settings = AppSettings(
            personalBufferMinutes = AppSettings.DEFAULT_PERSONAL_BUFFER_MINUTES,
            safetyMarginMinutes = AppSettings.DEFAULT_SAFETY_MARGIN_MINUTES,
            notificationsEnabled = AppSettings.DEFAULT_NOTIFICATIONS_ENABLED,
            defaultTransportMode = TransportMode.TRANSIT,
        )

        assertEquals(6, settings.personalBufferMinutes)
        assertEquals(5, settings.safetyMarginMinutes)
        assertFalse(settings.notificationsEnabled)
        assertEquals(TransportMode.TRANSIT, settings.defaultTransportMode)
    }

    @Test
    fun isValidBufferMinutes_acceptsOnlyZeroToSixty() {
        assertTrue(AppSettings.isValidBufferMinutes(0))
        assertTrue(AppSettings.isValidBufferMinutes(60))

        assertFalse(AppSettings.isValidBufferMinutes(-1))
        assertFalse(AppSettings.isValidBufferMinutes(61))
    }

    @Test
    fun bufferMinutesOrDefault_fallsBackForMissingOrInvalidValues() {
        assertEquals(12, AppSettings.bufferMinutesOrDefault(12, 6))
        assertEquals(6, AppSettings.bufferMinutesOrDefault(null, 6))
        assertEquals(6, AppSettings.bufferMinutesOrDefault(90, 6))
    }

    @Test
    fun transportModeOrDefault_fallsBackForUnknownNames() {
        assertEquals(TransportMode.WALK, AppSettings.transportModeOrDefault("WALK"))
        assertEquals(TransportMode.TRANSIT, AppSettings.transportModeOrDefault("BUS"))
        assertEquals(TransportMode.TRANSIT, AppSettings.transportModeOrDefault(null))
    }
}
